cookiecutter unit test failures with Click 7.0

* Cookiecutter version: 1.6.0 (latest master branch)
* Template project url: N/A
* Python version: Any version supported by Cookiecutter
* Operating System: Tested on macOS

### Description:

When running cookiecutter unit tests using the latest master branch, there are two test failures that occur with the latest version of Click (7.0). These failures do not occur with the previous version of Click (6.7). The failures happen with any of the Python versions supported by cookiecutter.

Travis-CI tests of the latest master branch are currently passing because they were last run 11 days ago, prior to the Click 7.0 release. I suspect rerunning Travis-CI will reproduce the failing tests described below.

The two failing tests assert usage error messages raised by Click. In Click 7.0, the argument's `metavar` parameter is displayed in the error message instead of the argument's `name` (https://github.com/pallets/click/pull/612). It appears that the metavar parameter defaults to the argument's uppercased name.

Fixing the tests should be straightforward to support Click 7.0 in addition to older versions. One potential fix is to assert that the *case-insensitive* error message matches what's expected. I'm happy to issue a PR to fix the tests.

### What I've run:

For example, running the unit tests with Python 2.7 and Click 7.0: 

```bash
$ tox -e py27
GLOB sdist-make: /Users/jairideout/dev/cookiecutter/setup.py
py27 inst-nodeps: /Users/jairideout/dev/cookiecutter/.tox/dist/cookiecutter-1.6.0.zip
py27 installed: arrow==0.12.1,atomicwrites==1.2.1,attrs==18.2.0,backports.functools-lru-cache==1.5,binaryornot==0.4.4,certifi==2018.8.24,chardet==3.0.4,Click==7.0,cookiecutter==1.6.0,coverage==4.5.1,freezegun==0.3.10,funcsigs==1.0.2,future==0.16.0,idna==2.7,Jinja2==2.10,jinja2-time==0.2.0,MarkupSafe==1.0,mock==2.0.0,more-itertools==4.3.0,pathlib2==2.3.2,pbr==4.2.0,pluggy==0.7.1,poyo==0.4.1,py==1.6.0,pytest==3.8.1,pytest-catchlog==1.2.2,pytest-cov==2.6.0,pytest-mock==1.10.0,python-dateutil==2.7.3,requests==2.19.1,scandir==1.9.0,six==1.11.0,urllib3==1.23,whichcraft==0.4.1
py27 run-test-pre: PYTHONHASHSEED='3751699627'
py27 runtests: commands[0] | pytest --cov=cookiecutter tests
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 2.7.10, pytest-3.8.1, py-1.6.0, pluggy-0.7.1
rootdir: /Users/jairideout/dev/cookiecutter, inifile: setup.cfg
plugins: mock-1.10.0, cov-2.6.0, catchlog-1.2.2
collected 261 items

tests/test_abort_generate_on_hook_error.py ..                                                                                                                                                        [  0%]
tests/test_cli.py ........................F..                                                                                                                                                        [ 11%]
tests/test_cookiecutter_invocation.py F.                                                                                                                                                             [ 11%]
tests/test_cookiecutter_local_no_input.py .......                                                                                                                                                    [ 14%]
tests/test_cookiecutter_local_with_input.py ..                                                                                                                                                       [ 15%]
tests/test_custom_extensions_in_hooks.py ..                                                                                                                                                          [ 16%]
tests/test_default_extensions.py .                                                                                                                                                                   [ 16%]
tests/test_environment.py ..                                                                                                                                                                         [ 17%]
tests/test_exceptions.py .                                                                                                                                                                           [ 17%]
tests/test_find.py ..                                                                                                                                                                                [ 18%]
tests/test_generate_context.py ..........                                                                                                                                                            [ 22%]
tests/test_generate_copy_without_render.py .                                                                                                                                                         [ 22%]
tests/test_generate_file.py .....                                                                                                                                                                    [ 24%]
tests/test_generate_files.py ..................                                                                                                                                                      [ 31%]
tests/test_generate_hooks.py ........                                                                                                                                                                [ 34%]
tests/test_get_config.py .....                                                                                                                                                                       [ 36%]
tests/test_get_user_config.py .........                                                                                                                                                              [ 39%]
tests/test_hooks.py ..........                                                                                                                                                                       [ 43%]
tests/test_log.py ...                                                                                                                                                                                [ 44%]
tests/test_main.py ..                                                                                                                                                                                [ 45%]
tests/test_output_folder.py ..                                                                                                                                                                       [ 46%]
tests/test_preferred_encoding.py s                                                                                                                                                                   [ 46%]
tests/test_prompt.py .........................                                                                                                                                                       [ 56%]
tests/test_read_repo_password.py .                                                                                                                                                                   [ 56%]
tests/test_read_user_choice.py .....                                                                                                                                                                 [ 58%]
tests/test_read_user_dict.py .......                                                                                                                                                                 [ 61%]
tests/test_read_user_variable.py .                                                                                                                                                                   [ 61%]
tests/test_read_user_yes_no.py .                                                                                                                                                                     [ 62%]
tests/test_repo_not_found.py .                                                                                                                                                                       [ 62%]
tests/test_specify_output_dir.py ..                                                                                                                                                                  [ 63%]
tests/test_utils.py .........                                                                                                                                                                        [ 66%]
tests/replay/test_dump.py .....                                                                                                                                                                      [ 68%]
tests/replay/test_load.py ....                                                                                                                                                                       [ 70%]
tests/replay/test_replay.py ......                                                                                                                                                                   [ 72%]
tests/repository/test_abbreviation_expansion.py .......                                                                                                                                              [ 75%]
tests/repository/test_determine_repo_dir_clones_repo.py .....                                                                                                                                        [ 77%]
tests/repository/test_determine_repo_dir_finds_existing_cookiecutter.py .                                                                                                                            [ 77%]
tests/repository/test_determine_repository_should_use_local_repo.py ...                                                                                                                              [ 78%]
tests/repository/test_is_repo_url.py .............                                                                                                                                                   [ 83%]
tests/repository/test_repository_has_cookiecutter_json.py ...                                                                                                                                        [ 84%]
tests/vcs/test_clone.py ..........                                                                                                                                                                   [ 88%]
tests/vcs/test_identify_repo.py .............                                                                                                                                                        [ 93%]
tests/vcs/test_is_vcs_installed.py ....                                                                                                                                                              [ 95%]
tests/zipfile/test_unzip.py .............                                                                                                                                                            [100%]

================================================================================================= FAILURES =================================================================================================
__________________________________________________________________________________ test_cli_extra_context_invalid_format ___________________________________________________________________________________

cli_runner = <function cli_main at 0x102f4d938>

    @pytest.mark.usefixtures('remove_fake_project_dir')
    def test_cli_extra_context_invalid_format(cli_runner):
        result = cli_runner(
            'tests/fake-repo-pre/',
            '--no-input',
            '-v',
            'ExtraContextWithNoEqualsSoInvalid',
        )
        assert result.exit_code == 2
>       assert 'Error: Invalid value for "extra_context"' in result.output
E       assert 'Error: Invalid value for "extra_context"' in 'Usage: main [OPTIONS] TEMPLATE [EXTRA_CONTEXT]...\n\nError: Invalid value for "[EXTRA_CONTEXT]...": EXTRA_CONTEXT should contain items of the form key=value; \'ExtraContextWithNoEqualsSoInvalid\' doesn\'t match that form\n'
E        +  where 'Usage: main [OPTIONS] TEMPLATE [EXTRA_CONTEXT]...\n\nError: Invalid value for "[EXTRA_CONTEXT]...": EXTRA_CONTEXT should contain items of the form key=value; \'ExtraContextWithNoEqualsSoInvalid\' doesn\'t match that form\n' = <Result SystemExit(2,)>.output

tests/test_cli.py:393: AssertionError
_______________________________________________________________________________ test_should_raise_error_without_template_arg _______________________________________________________________________________

capfd = <_pytest.capture.CaptureFixture object at 0x103b2f7d0>

    def test_should_raise_error_without_template_arg(capfd):
        with pytest.raises(subprocess.CalledProcessError):
            subprocess.check_call(['python', '-m', 'cookiecutter.cli'])

        _, err = capfd.readouterr()
        exp_message = 'Error: Missing argument "template".'
>       assert exp_message in err
E       assert 'Error: Missing argument "template".' in 'Usage: cli.py [OPTIONS] TEMPLATE [EXTRA_CONTEXT]...\nTry "cli.py -h" for help.\n\nError: Missing argument "TEMPLATE".\n'

tests/test_cookiecutter_invocation.py:25: AssertionError

---------- coverage: platform darwin, python 2.7.10-final-0 ----------
Name                          Stmts   Miss  Cover
-------------------------------------------------
cookiecutter/__init__.py          2      0   100%
cookiecutter/__main__.py          3      0   100%
cookiecutter/cli.py              50      0   100%
cookiecutter/config.py           52      0   100%
cookiecutter/environment.py      21      0   100%
cookiecutter/exceptions.py       24      0   100%
cookiecutter/extensions.py        9      0   100%
cookiecutter/find.py             18      0   100%
cookiecutter/generate.py        166      0   100%
cookiecutter/hooks.py            61      0   100%
cookiecutter/log.py              22      0   100%
cookiecutter/main.py             31      0   100%
cookiecutter/prompt.py           90      0   100%
cookiecutter/replay.py           30      0   100%
cookiecutter/repository.py       39      0   100%
cookiecutter/utils.py            50      2    96%
cookiecutter/vcs.py              54      0   100%
cookiecutter/zipfile.py          61      0   100%
-------------------------------------------------
TOTAL                           783      2    99%

============================================================================================= warnings summary =============================================================================================
  pytest-catchlog plugin has been merged into the core, please remove it from your requirements.

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================================================= 2 failed, 258 passed, 1 skipped, 1 warnings in 4.76 seconds ========================================================================
ERROR: InvocationError for command '/Users/jairideout/dev/cookiecutter/.tox/py27/bin/pytest --cov=cookiecutter tests' (exited with code 1)
_________________________________________________________________________________________________ summary __________________________________________________________________________________________________
ERROR:   py27: commands failed
```

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -390,7 +390,7 @@ def test_cli_extra_context_invalid_format(cli_runner):
         'ExtraContextWithNoEqualsSoInvalid',
     )
     assert result.exit_code == 2
-    assert 'Error: Invalid value for "extra_context"' in result.output
+    assert 'Error: Invalid value for "[EXTRA_CONTEXT]..."' in result.output
     assert 'should contain items of the form key=value' in result.output
 
 

--- a/tests/test_cookiecutter_invocation.py
+++ b/tests/test_cookiecutter_invocation.py
@@ -21,7 +21,7 @@ def test_should_raise_error_without_template_arg(capfd):
         subprocess.check_call(['python', '-m', 'cookiecutter.cli'])
 
     _, err = capfd.readouterr()
-    exp_message = 'Error: Missing argument "template".'
+    exp_message = 'Error: Missing argument "TEMPLATE".'
     assert exp_message in err
 
 

